tools/autograd/derivatives.yaml

```diff
@@ -669,6 +669,10 @@
   self: not_implemented("_pdist_backward")
   pdist: not_implemented("_pdist_backward")
 
+- name: _euclidean_dist(Tensor x1, Tensor x2) -> Tensor
+  x1: _euclidean_dist_backward(grad, x1, x2, 0, result)
```

Comment:
this can be slightly simpler if you use a single function to do both: defining it like [this](https://github.com/pytorch/pytorch/blob/4ed790d74223ff3af95bbf8585cb67664805b6b9/tools/autograd/derivatives.yaml#L193-L194) as `x1, x2: xxx` and making your function return a tuple like [this](https://github.com/pytorch/pytorch/blob/4ed790d74223ff3af95bbf8585cb67664805b6b9/tools/autograd/templates/Functions.cpp#L1626-L1631).